src/core_modules/capture-core-utils/rulesEngine/services/VariableService/VariableService.js

```diff
@@ -206,7 +206,7 @@ export class VariableService {
         return {
             variableValue: this.onProcessValue(value, type),
             useCodeForOptionSet: !useNameForOptionSet,
-            variableType: useNameForOptionSet ? typeKeys.TEXT : type,
+            variableType: type || typeKeys.TEXT,
```

Comment:
Seems like a bold change to exclude `useNameForOptionSet` from the expression entirely, but maybe you see the reason for it being there in the first place?

Reply:
This was an error actually, but its fixed by hard coding "TEXT" outside of the call to this function when there IS an option set AND nameForOptionSet == true. Fixed in this commit: https://github.com/dhis2/capture-app/pull/2560/commits/d0e00d779d5928b6b5528cee5fd903e98dd3396c